app/controllers/merchant_invoice_items_controller.rb

```diff
@@ -0,0 +1,20 @@
+class MerchantInvoiceItemsController < ApplicationController
+  def update
+    @invoice_item = InvoiceItem.find(params[:id])
+
+    if params[:status] != @invoice_item.status
+      @invoice_item.update(invoice_item_params)
+      redirect_to merchant_invoice_path(@invoice_item.merchant, @invoice_item.invoice)
+    else
+      redirect_to merchant_invoice_path(@invoice_item.merchant, @invoice_item.invoice)
+      flash[:alert] = "Status must be changed"
+    end
+  end
+
```

Comment:
Good work on the update logic. Nice redirect and just overall logic of the update action.